distribution/docker/alpine/Dockerfile

```diff
@@ -5,11 +5,14 @@ WORKDIR /vector
 COPY vector-*-unknown-linux-musl*.tar.gz ./
 RUN tar -xvf vector-0*-$(cat /etc/apk/arch)-unknown-linux-musl*.tar.gz --strip-components=2
 
+RUN mkdir -p /var/lib/vector
+
 FROM alpine:3.14
 RUN apk update && apk add ca-certificates tzdata && rm -rf /var/cache/apk/*
 
 COPY --from=builder /vector/bin/* /usr/local/bin/
 COPY --from=builder /vector/config/vector.toml /etc/vector/vector.toml
+COPY --from=builder /var/lib/vector /var/lib/vector
```

Comment:
Is this step needed? I don't think there'd actually be anything to copy.

Reply:
It is needed because I create the empty directory in a previous stage (`builder`) where `mkdir` is guaranteed to be available. Then, I `COPY` the empty directory from the `builder` stage to the final stage where `mkdir` might not be available, notably in the distroless-variant images.

Not all variants lack `mkdir` in the final image, so I could have skipped the `COPY` step on those, true, but I preferred to make consistent changes that are independent of the image variant. Thus (I think) making them easier to read/maintain.

If you prefer me to make more tailored changes for each variant, let me know and it can be done like that too.